### PR TITLE
Adding myst-parser to jedi-tools-env

### LINKED ISSUE #234: Add Sphinx MyST-parser to spack and the to spack-stack
The `jedi-tools-env` needs package `myst-parser` for Sphinx to build the JEDI documentation (not needed for spack -stack documentation). This package doesn't exist yet in spack, therefore need to add it first in spack, and then make any necessary updates in spack-stack.

### PR DESCRIPTION
**Pull request by ashley314**
## Description

myst-parser is needed for Sphinx to build the JEDI documentation

## Issue(s) addressed

Resolves https://github.com/JCSDA/spack-stack/issues/234

## Dependencies

None

## Impact

None

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have run the unit tests before creating the PR

### CODE REVIEW

**Comment by ashley314** [comment]
Successfully tested on my mac. 
```
EIT-L-18153:docs agriffin$ make html
Running Sphinx v5.3.0
making output directory... done
checking bibtex cache... out of date
parsing bibtex file /Users/agriffin/JEDI/jedi-docs/docs/ref/refs.bib... parsed 10 entries
myst v0.18.1: MdParserConfig(commonmark_only=False, gfm_only=False, enable_extensions=[], disable_syntax=[], all_links_external=False, url_schemes=('http', 'https', 'mailto', 'ftp'), ref_domains=None, highlight_code_blocks=True, number_code_blocks=[], title_to_header=False, heading_anchors=None, heading_slug_func=None, footnote_transition=True, words_per_minute=200, sub_delimiters=('{', '}'), linkify_fuzzy_links=True, dmath_allow_labels=True, dmath_allow_space=True, dmath_allow_digits=True, dmath_double_inline=False, update_mathjax=True, mathjax_classes='tex2jax_process|mathjax_process|math|output_area')
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 222 source files that are out of date
updating environment: [new config] 222 added, 0 changed, 0 removed
reading sources... [100%] working-practices/testing                                                                                                          
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
writing output... [100%] working-practices/testing                                                                                                           
generating indices... genindex done
writing additional pages... search done
copying images... [100%] inside/testing/images/Test_class.png                                                                                                
copying static files... done
copying extra files... done
dumping search index in English (code: en)... done
dumping object inventory... done
build succeeded.
```

File: var/spack/repos/jcsda-emc-bundles/packages/jedi-tools-env/package.py [hunk 1]
```diff
@@ -21,15 +21,16 @@ class JediToolsEnv(BundlePackage):
...
+    # depends_on("awscli", type="run")
+    # depends_on("aws-parallelcluster", type="run")
     depends_on("py-click", type="run")
     depends_on("py-openpyxl", type="run")
     depends_on("py-pandas", type="run")
     depends_on("py-pygithub", type="run")
     depends_on("py-scipy", type="run")
     depends_on("py-sphinx", type="run")
-    depends_on("py-sphinxcontrib-bibtex", when="+latex", type="run")
+    depends_on("py-myst-parser", type="run")
```
Comment (climbfuji) [inline var/spack/repos/jcsda-emc-bundles/packages/jedi-tools-env/package.py:32]: I took a quick look in the spack-stack configs folder and couldn't see any place where we enable/disable this variant. But you need to remove the variant entirely (lines 19 and 20).

**Review by climbfuji — CHANGES_REQUESTED** [review]

File: var/spack/repos/jcsda-emc-bundles/packages/jedi-tools-env/package.py (hunk 1 above)
Comment (ashley314) [reply var/spack/repos/jcsda-emc-bundles/packages/jedi-tools-env/package.py:32]: @climbfuji what about the mention in line 34?

Comment (climbfuji) [reply var/spack/repos/jcsda-emc-bundles/packages/jedi-tools-env/package.py:32]: Oh, I see. You are correct, the variant needs to stay for that line. Sorry!

**Review by srherbener — APPROVED** [review]
Looks good. Thanks!

**Review by climbfuji — APPROVED** [review]

**Comment by srherbener** [comment]
@climbfuji after merging this PR should we make a corresponding "update the submodule commit hash" in spack-stack or should we just wait for another PR set to sweep this in?

I can do the spack-stack PR if that's what we should do.

Thanks!

**Comment by climbfuji** [comment]
> @climbfuji after merging this PR should we make a corresponding "update the submodule commit hash" in spack-stack or should we just wait for another PR set to sweep this in?
> 
> I can do the spack-stack PR if that's what we should do.
> 
> Thanks!

Would be good to have a spack-stack PR before merging to test it, please go ahead if you have time!

**Pull request merged**